File: 2019/jul21/beauty.cpp
#include<iostream>
#include<string>

using namespace std;
int main(int argc, char const *argv[])
{
    int t;
    cin>>t;
    while(t--)
    {
        string s;
        cin>>s;
        int count=0;
        int lenght=s.length();
        for(int i=0;i<lenght;i++)
        {
            for(int j=1;j<=lenght-i;j++)
            {
                string temp=s.substr(i,j);
                // cout<<temp<<endl;
                int count0=0;
                int count1=0;
                int tl=temp.length();
                for(int k=i;k<=tl;k++)
                {
                    if(temp[k]=='0')
                    {
                        count0++;
                    }
                    else if(temp[k]=='1')
                    {
                        count1++;
                    }
                    
                }
                if(count0==(count1*count1))
                {
                    // cout<<count0<<" "<<count1<<" "<<temp<<" "<<count<<endl;
                    count++;
                }
                // cout<<endl;

            }
        }
        cout<<count<<endl;
        
        

    }

    return 0;
}

File: 2019/jul21/beauty.java
import java.util.*;

public class beauty
{
    public static void main(String[] args) 
    {
        int t;
        Scanner scn=new Scanner(System.in);
        t=scn.nextInt();
        while(t>0)
        {
            String s;
            s=scn.nextLine();
            int count=0;
            for (int i = 0; i < s.length(); i++) 
            {
                for (int j = i+1; j <= s.length(); j++) 
                {
                      String temp=s.substring(i,j);
                      int tl=temp.length();
                      int count0=0,count1=0;

                      for(int k=0;k<tl;k++)
                      {
                          if(temp.charAt(k)=='0')
                          {
                              count0++;
                          }
                          else if(temp.charAt(k)=='1')
                          {
                              count1++;

                          }
                      }
                      if(count0==(count1*count1))
                      {
                          count++;
                      }

               
                }
            }
            System.out.println(count);




            t--;

        }
        scn.close();
    }
}